hivemind/utils/mpfuture.py

```diff
@@ -63,36 +83,49 @@ class MPFuture(base.Future, Generic[ResultType]):
 
     _initialization_lock = mp.Lock()  # global lock that prevents simultaneous initialization of two processes
     _update_lock = mp.Lock()  # global lock that prevents simultaneous writing to the same pipe
-    _process_wide_pipe: Optional[PipeEnd] = None  # a pipe that is used to send results/exceptions to this process
+    _global_sender_pipe: Optional[PipeEnd] = None  # a pipe that is used to send results/exceptions to this process
     _pipe_waiter_thread: Optional[threading.Thread] = None  # process-specific thread that receives results/exceptions
     _active_futures: Optional[Dict[UID, Type[ref][MPFuture]]] = None  # non-done futures originated from this process
```

Comment:
```suggestion
    _active_futures: Optional[Dict[UID, 'ref[MPFuture]']] = None  # non-done futures originated from this process
```

`ref[MPFuture]` is made syntactically correct in Python 3.9 (as well as `asyncio.Future[int]`) but we need quotes for earlier Python versions.